lib/csu/aarch64/crt1_c.c

```diff
@@ -77,21 +77,16 @@ __start(int argc, char *argv[], char *env[], void (*cleanup)(void))
 	 * would initialize them if they did.
 	 */
 	if (&_DYNAMIC == NULL) {
-		const Elf_Phdr * __capability phdr_cap = NULL;
 		const Elf_Auxinfo *auxp;
 		char **strp;
 		void *phdr = NULL;
 		long phnum = 0;
 
 		strp = env;
-		while (*strp == NULL)
-			strp++;
-		auxp = (Elf_Auxinfo *)(strp + 1);
-
-		/*
-		 * Digest the auxiliary vector to get information
-		 * needed to init globals.
-		 */
+		while (*strp++ != NULL)
```

Comment:
The commit message says "moves the increment to happen before the comparison" which I find confusing given this is a post increment. I'd say something about the increment being unconditional instead.

Reply:
Reworded; hopefully better now?

Reply:
Or just "however this approach matches the style in init_aux_vector_once".

Reply:
works for me